### PR TITLE
Two patches to make the generated headers reproducible

### PR DESCRIPTION
**Pull request by jspricke**

### CODE REVIEW

File: scripts/msg.h.template [hunk 1]
```diff
@@ -183,26 +184,26 @@ namespace message_traits
...
+// !!!!!!!!!!! @(sorted(dir(spec)))
 @#if spec.is_fixed_length():
 @#    traits = traits.append('IsFixedSize')
 
 @#template <class ContainerAllocator>
-@#struct IsFixedSize<@(cpp_full_name_with_alloc) >:: @(dir(spec)) ? "TrueType" ! "FalseType") { };
+@#struct IsFixedSize<@(cpp_full_name_with_alloc) >:: @(sorted(dir(spec))) ? "TrueType" ! "FalseType") { };
 @#template <class ContainerAllocator>
-@#struct IsFixedSize<@(cpp_full_name_with_alloc) const >:: @(dir(spec)) ? "TrueType" ! "FalseType") { };
+@#struct IsFixedSize<@(cpp_full_name_with_alloc) const >:: @(sorted(dir(spec))) ? "TrueType" ! "FalseType") { };
```
Comment (dirk-thomas) [inline scripts/msg.h.template]: All of these are only template comments which are never part of the generated header. Reading through the comments it seems all of this belongs under the `TODO` from line 197/198 but has been implemented already below.

How about deleting all lines starting from `// BOOLTRAITS @bool_traits` until this line? Neither seems to contains any useful information in the generated header.

Comment (jspricke) [reply scripts/msg.h.template]: I haven't looked into how empy works, but at least the `IsFixedSize` seem to generate method definitions in the generated header. I would be fine with deleting from BOOLTRAITS to !!!!!.

Comment (dirk-thomas) [reply scripts/msg.h.template]: > I haven't looked into how empy works

Anything behind `@#` is a comment and won't appear in the expanded template.

> but at least the `IsFixedSize` seem to generate method definitions in the generated header.

The corresponding template snippet is in the for loop below.

Comment (jspricke) [reply scripts/msg.h.template]: Done, thanks for the heads up :).

**Comment by dirk-thomas** [comment]
Thanks for the patch / update.

**Pull request merged**